File: English/Stream/Go/GPT/internal/models/song.go
package models

import "time"

type Song struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Genre       string    `json:"genre"`
	Description string    `json:"description,omitempty"`
	FilePath    string    `json:"-"`
	ArtistID    string    `json:"artistId"`
	ArtistName  string    `json:"artistName"`
	CreatedAt   time.Time `json:"createdAt"`
}

type SongFilter struct {
	Title  string
	Artist string
	Genre  string
}

File: English/Stream/Python/GPT/backend/app/models/song.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class Song(Base):
    __tablename__ = "songs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    title: Mapped[str] = mapped_column(String(255), index=True, nullable=False)
    genre: Mapped[str] = mapped_column(String(120), index=True, nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    artist_id: Mapped[str] = mapped_column(String(120), index=True, nullable=False)
    artist_name: Mapped[str] = mapped_column(String(255), index=True, nullable=False)
    file_path: Mapped[str] = mapped_column(String(500), nullable=False, unique=True)
    original_file_name: Mapped[str] = mapped_column(String(255), nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)

